lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -290,10 +291,12 @@ sub start_domain_test {
     my $result = 0;
 
     $params->{domain} =~ s/^\.// unless ( !$params->{domain} || $params->{domain} eq '.' );
+    $params->{domain} .= '.' if '.' ne substr $params->{domain}, -1;
+
     my $syntax_result = $self->validate_syntax( $params );
     die "$syntax_result->{message} \n" unless ( $syntax_result && $syntax_result->{status} eq 'ok' );
 
-    die "No domain in parameters\n" unless ( $params->{domain} );
+    die "No domain in parameters\n" unless ( $params->{domain} ); #Dead code ?
```

Comment:
Isn't it better to die here rather than risk of lettering empty domain go through?

Reply:
We can keep it, but five lines before  `$params->{domain} =~ s/^\.// unless ( !$params->{domain} || $params->{domain} eq '.' );` we set the value of the empty domain to '.'.

Reply:
The code for normalizing the domain argument is a bit messy though. Maybe it could be rewritten so that it's obvious from a glance that the domain argument is non-empty?
Otherwise maybe the assertion could be placed together with the normalization and the message could be updated to something like "Domain should be non-empty".

Reply:
I find it strange to set the the value of an empty domain to '.'. Empty domain is an error, not the same as the root zone.